tests/test_parser.py

```diff
@@ -380,3 +454,46 @@ def test_tls_help(self):
         self.parser._set_defaults(None)
         help_output = self.parser._parser_tls.format_help()
         self.assert_snapshot('tls_help', help_output)
+
+
+class ParserModuleFunctionTestCase(unittest.TestCase):
```

Comment:
Personally I would split this into two different test cases. One for `create_parser` and the other for `create_connection`. It might look as wasting code but it improves separation of concerns.

Reply:
That's a great idea!